### PR TITLE
Adding KubernetesPodOperator example for Composer 2

### PR DESCRIPTION
**Pull request by rafalbiegacz**
## Description

Adding KubernetesPodOperator example for Composer 2
- It updates namespace configuration to use default one configured in Composer 2
- properly configures requests and limits for Autopilot clusters
- defines kubernetes_conn_id and config_file
- removes a task related to anti-affinity as it's yet supported for Composer 2

## Checklist
- [x] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [x] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [ ] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [x] Please **merge** this PR for me once it is approved.
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample

### CODE REVIEW

File: composer/workflows/c2-kubernetes_pod_operator.py [hunk 1]
```diff
@@ -0,0 +1,217 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the {{ quot_code }}License{{ quot_code }});
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an {{ quot_code }}AS IS{{ quot_code }} BASIS,
```
Comment (leahecole) [inline composer/workflows/c2-kubernetes_pod_operator.py:10]:
```suggestion
# distributed under the License is distributed on an "AS IS" BASIS,
```

File: composer/workflows/c2-kubernetes_pod_operator.py [hunk 2]
```diff
@@ -0,0 +1,217 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the {{ quot_code }}License{{ quot_code }});
```
Comment (leahecole) [inline composer/workflows/c2-kubernetes_pod_operator.py:3]:
```suggestion
# Licensed under the Apache License, Version 2.0 (the "License");
```

File: composer/workflows/c2-kubernetes_pod_operator.py [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an {{ quot_code }}AS IS{{ quot_code }} BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# pylint: disable=bad-indentation
```
Comment (leahecole) [inline composer/workflows/c2-kubernetes_pod_operator.py:15]:
```suggestion
```

File: composer/workflows/c2-kubernetes_pod_operator.py [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+    KubernetesPodOperator,
+)
+from kubernetes.client import models as k8s_models
+
+# A Secret is an object that contains a small amount of sensitive data such as
+# a password, a token, or a key. Such information might otherwise be put in a
+# Pod specification or in an image; putting it in a Secret object allows for
+# more control over how it is used, and reduces the risk of accidental
+# exposure.
+
```
Comment (leahecole) [inline composer/workflows/c2-kubernetes_pod_operator.py:32]:
```suggestion
# [START composer_2_kubernetespodoperator_secretobject]
```

File: composer/workflows/c2-kubernetes_pod_operator.py [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+secret_volume = Secret(
+    deploy_type="volume",
+    # Path where we mount the secret as volume
+    deploy_target="/var/secrets/google",
+    # Name of Kubernetes Secret
+    secret="service-account",
+    # Key in the form of service account file name
+    key="service-account.json",
+)
+# If you are running Airflow in more than one time zone
```
Comment (leahecole) [inline composer/workflows/c2-kubernetes_pod_operator.py:53]:
```suggestion
# [END composer_2_kubernetespodoperator_secretobject]
# If you are running Airflow in more than one time zone
```

File: composer/workflows/c2-kubernetes_pod_operator.py [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an {{ quot_code }}AS IS{{ quot_code }} BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# pylint: disable=bad-indentation
+
```
Comment (leahecole) [inline composer/workflows/c2-kubernetes_pod_operator.py:16]:
```suggestion
# [START composer_2_kubernetespodoperator]
```

File: composer/workflows/c2-kubernetes_pod_operator.py [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+        kubernetes_conn_id="kubernetes_default",
+        # Affinity determines which nodes the Pod can run on based on the
+        # config. For more information see:
+        # https://kubernetes.io/docs/concepts/configuration/assign-pod-node/
+        # Pod affinity with the KubernetesPodOperator
+        # is not supported with Composer 2
+        # instead, create a cluster and use the GKEStartPodOperator
+        # https://cloud.google.com/composer/docs/using-gke-operator
+        affinity={},
+    )
```
Comment (leahecole) [inline composer/workflows/c2-kubernetes_pod_operator.py:217]:
```suggestion
    )
    # [END composer_2_kubernetespodoperator_fullconfig]
    # [END composer_2_kubernetespodoperator]

```

File: composer/workflows/c2-kubernetes_pod_operator.py [hunk 8]
```diff
@@ -0,0 +1,217 @@
...
+    start_date=YESTERDAY,
+) as dag:
+    # Only name, namespace, image, and task_id are required to create a
+    # KubernetesPodOperator. In Cloud Composer, currently the operator defaults
+    # to using the config file found at `/home/airflow/composer_kube_config if
+    # no `config_file` parameter is specified. By default it will contain the
+    # credentials for Cloud Composer's Google Kubernetes Engine cluster that is
+    # created upon environment creation.
+
+    kubernetes_min_pod = KubernetesPodOperator(
```
Comment (leahecole) [inline composer/workflows/c2-kubernetes_pod_operator.py:73]:
```suggestion
    # [START composer_2_kubernetespodoperator_minconfig]
    kubernetes_min_pod = KubernetesPodOperator(
```

File: composer/workflows/c2-kubernetes_pod_operator.py [hunk 9]
```diff
@@ -0,0 +1,217 @@
...
+        # uses has permission to access the Google Container Registry
+        # (the default service account has permission)
+        image="gcr.io/gcp-runtimes/ubuntu_20_0_4",
+        # Specifies path to kubernetes config. If no config is specified will
+        # default to '~/.kube/config'. The config_file is templated.
+        config_file="/home/airflow/composer_kube_config",
+        # Identifier of connection that should be used
+        kubernetes_conn_id="kubernetes_default",
+    )
+    kubernetes_template_ex = KubernetesPodOperator(
```
Comment (leahecole) [inline composer/workflows/c2-kubernetes_pod_operator.py:103]:
```suggestion
    # [END composer_2_kubernetespodoperator_minconfig]
    # [START composer_2_kubernetespodoperator_templateconfig]
    kubernetes_template_ex = KubernetesPodOperator(
```

File: composer/workflows/c2-kubernetes_pod_operator.py [hunk 10]
```diff
@@ -0,0 +1,217 @@
...
+            "EXAMPLE_VAR": "/example/value",
+            "GOOGLE_APPLICATION_CREDENTIALS": "/var/secrets/google/service-account.json ",
+        },
+        # Specifies path to kubernetes config. If no config is specified will
+        # default to '~/.kube/config'. The config_file is templated.
+        config_file="/home/airflow/composer_kube_config",
+        # Identifier of connection that should be used
+        kubernetes_conn_id="kubernetes_default",
+    )
+    kubernetes_full_pod = KubernetesPodOperator(
```
Comment (leahecole) [inline composer/workflows/c2-kubernetes_pod_operator.py:153]:
```suggestion
    # [END composer_2_kubernetespodoperator_secretconfig]
    # [START composer_2_kubernetespodoperator_fullconfig]
    kubernetes_full_pod = KubernetesPodOperator(
```

**Review by leahecole — CHANGES_REQUESTED** [review]
- Please rename the files to kubernetes_pod_operator_c2.py and kubernetes_pod_operator_c2_test.py - that way they'll show up grouped with the other KPO file. This should also fix your test failure
- Please remove the pylint disable and run "black" on your code. This can be done via the nox session and you can also run lint locally with a nox session (more info on that in the [Authoring guide](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md) - ping me if you need help) - I recommend doing this AFTER adding in my suggestions just in case I accidentally introduce a whitespace error 😅 
- Please accept the suggestions that add in region tags - @pavel-salnikov will need these so that we can properly embed the code in the docs

**Review by pavel-salnikov — APPROVED** [review]

**Review by leahecole — APPROVED** [review]

**Pull request merged**